### ISSUE TITLE
Escape filenames when copying from S3

### ISSUE THREAD

**Issue by sendivogius** [nextflow-io/nextflow#717]
When executing process (with awsbatch executor) which downloads files from S3 file names are not escaped.

For example processing file '01_A(R1).fastq' results in following content of command.run:

```
# stage input files
rm -f 01_A\(R1\).fastq
rm -f 01_A\(R2\).fastq
aws s3 cp --only-show-errors s3://some_bucket/01_A(R1).fastq 01_A(R1).fastq
aws s3 cp --only-show-errors s3://some_bucket/01_A(R2).fastq 01_A(R2).fastq
```

Tested on version:
N E X T F L O W  ~  version 0.28.1

**Comment by pditommaso** [comment]
Pushed a snapshot that should solve the problem. Please give it a try using the following command: 

```
NXF_VER=0.30.0-SNAPSHOT nextflow run .. etc
```

**Comment by sendivogius** [comment]
Files are now correctly downloaded from S3.
However I have now a problem with uploading files to S3. I got message that output file is missing but I can see it when I run ls inside process.


```
Caused by:
  Missing output file(s) `01(A)_annotated.vcf` expected by process `get_annotations (01(A))`

Command executed:

  python /get_annotations.py --vcf in.vcf --output '01(A)_annotated'
  ls -l

Command exit status:
  0

Command output:
  File successfully processed and result saved to: 01(A)_annotated.vcf and 01(A)_annotated.tsv
  -rw-r--r-- 1 root root 19615431 May 24 11:05 01(A)_annotated.tsv
  -rw-r--r-- 1 root root 17716743 May 24 11:05 01(A)_annotated.vcf
  -rw-r--r-- 1 root root 21854664 May 24 11:01 in.vcf

Work dir:
  s3://test_bucket/batch_workdir/d6/ba5edd777ca121db5598e81a3a6ac9

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`

 -- Check '.nextflow.log' file for details
```

**Comment by pditommaso** [comment]
Ouch! Uploaded a new snapshot `0.30.0-SNAPSHOT build 4805` that should fix this problem as well. 

To try it refresh your snapshot with this command:

```
CAPSULE_RESET=1 NXF_VER=0.30.0-SNAPSHOT nextflow info
```

Then run nextflow as before: 

```
NXF_VER=0.30.0-SNAPSHOT nextflow run .. etc
```

**Comment by sendivogius** [comment]
It is working now for filenames without spaces, but there is still problem when filename contains space.
After quick investigation I localized problem to `for` loop in `nxf_s3_upload` function - by default values in loop are separated by spaces.
The solution might be to change value of IFS variable.

**Comment by pditommaso** [comment]
Ugly, are you able to propose a patch? (note that code should be able to resolve wildcards correctly)

**Comment by pditommaso** [comment]
OK, I've uploaded a new patch that fix also the problem with blanks. 

As before to try it refresh your snapshot with this command:

```
CAPSULE_RESET=1 NXF_VER=0.30.0-SNAPSHOT nextflow info
```

Then run nextflow as before: 

```
NXF_VER=0.30.0-SNAPSHOT nextflow run .. etc
```

**Comment by pditommaso** [comment]
I've made some test and it's working file. It's available in version `0.30.0-SNAPSHOT`.
